### PR TITLE
Change file create mode to 0600

### PR DESCRIPTION
**Pull request by patrickxb**

### CODE REVIEW

**Comment by patrickxb** [comment]
You mean the result of MakePreview?  

Don't you move the preview file into place with conversationid-msgid in the filename once you get the message ID?  It's only supposed to be temporary to enhance the UI for the user uploading an image.

**Comment by patrickxb** [comment]
(the above code is for downloading previews and full files)

File: go/service/chat_local.go [hunk 1]
```diff
@@ -1385,7 +1385,7 @@ func (h *chatLocalHandler) DownloadFileAttachmentLocal(ctx context.Context, arg
 		Preview:          arg.Preview,
 		IdentifyBehavior: arg.IdentifyBehavior,
 	}
-	sink, err := os.Create(arg.Filename)
+	sink, err := os.OpenFile(arg.Filename, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```
Comment (oconnor663) [inline go/service/chat_local.go]: https://golang.org/pkg/io/ioutil/#TempFile might make this simpler?

**Comment by patrickxb** [comment]
No. Frontend provides the filename.

TempFile is used by MakePreview.

**Comment by cjb** [comment]
Oh, wait.  This doesn't fix all of the problem.  The filename still contains the title/filename of the uploaded attachment, and is still world-readable in /tmp.

**Comment by cjb** [comment]
I guess my preference here would be for the service to manage creating `os.tmpDir + "/keybase_chat_files/"` with 0700 perms, and ignore any directory coming from the GUI and just take the GUI's filename.  We should probably talk about it more.

**Comment by patrickxb** [comment]
I'm merging this so we have 0600 regardless.  We can ticket anything further that needs to be done.

**Pull request merged**